File: solutions/go/lasagna-master/1/lasagna_master.go
package lasagna

// TODO: define the 'PreparationTime()' function
func PreparationTime(layer []string, minute int) int {
	if minute == 0 {
		minute = 2
	}
	numberLayer := len(layer)
	return minute * numberLayer
}

// TODO: define the 'Quantities()' function
func Quantities(layer []string) (int, float64) {
	var numberNoodles int
	var numberSauce int
	for _, v := range layer {
		switch v {
		case "noodles":
			numberNoodles++
		case "sauce":
			numberSauce++
		}
	}
	noodles := numberNoodles * 50
	sauce := float64(numberSauce) * 0.2
	return noodles, sauce
}

// TODO: define the 'AddSecretIngredient()' function
func AddSecretIngredient(friendsList []string, myList []string) {
	secretIngredient := friendsList[len(friendsList)-1]
	myList[len(myList)-1] = secretIngredient
}

// TODO: define the 'ScaleRecipe()' function
func ScaleRecipe(quantities []float64, portion int) []float64 {
	quantitiesCopy := make([]float64, len(quantities))
	copy(quantitiesCopy, quantities)
	var coefficient float64 = float64(portion) / 2.0
	for i, v := range quantitiesCopy {
		quantitiesCopy[i] = v * coefficient
	}
	return quantitiesCopy
}

// Your first steps could be to read through the tasks, and create
// these functions with their correct parameter lists and return types.
// The function body only needs to contain `panic("")`.
//
// This will make the tests compile, but they will fail.
// You can then implement the function logic one by one and see
// an increasing number of tests passing as you implement more
// functionality.

File: solutions/javascript/lasagna-master/1/lasagna-master.js
/// <reference path="./global.d.ts" />
// @ts-check

/**
 * Determine whether the lasagna is done
 *
 * @param {number} time
 *
 *
 *  @return {String} return a string with a information about the lasagna timer
 */
export function cookingStatus(time) {
  return time === 0
    ? "Lasagna is done."
    : time > 0
      ? "Not done, please wait."
      : "You forgot to set the timer.";
}

/**
 * Estimate the preparation time
 *
 * @param {Array<String>} layer
 * @param {?number} time
 *
 *  @return {number} return a number of time for preparation time
 */
export function preparationTime(layer, time) {
  return layer.length * (time ?? 2);
}

/**
 * Compute the amounts of noodles and sauce needed
 *
 * @param {Array<String>} ingredient
 *
 *  @return {object} return an object with the quantities of sauce and noodle needed
 */
export function quantities(ingredient) {
  let noodles = 0;
  let sauce = 0;
  for (let i = 0; i < ingredient.length; i++) {
    if (ingredient[i] === "noodles") {
      noodles += 50;
    }
    if (ingredient[i] === "sauce") {
      sauce += 0.2;
    }
  }
  return { noodles: noodles, sauce: sauce };
}

/**
 * Compute the amounts of noodles and sauce needed
 *
 * @param {Array<String>} friendsList
 * @param {Array<String>} myList
 *a
 */

export function addSecretIngredient(friendsList, myList) {
  myList.push(friendsList[friendsList.length - 1]);
}

/**
 * Scale the recipe
 *
 * @param {object} recipe
 * @param {number} portion
 *
 *  @return {object} return an object with the quantities of sauce and noodle needed
 */
export function scaleRecipe(recipe, portion) {
  let modifyRecipe = Object.assign({}, recipe);
  for (let key in modifyRecipe) {
    modifyRecipe[key] = (modifyRecipe[key] / 2) * portion;
  }
  return modifyRecipe;
}
